### PR TITLE
Reduce aggregate size of DB connection pools

### PR DESCRIPTION
**Pull request by rin-skylight**
## Related Issue or Background Info

Originally sourced from #2181. In short, we are seeing potentially inflated application latency and resource consumption due to a large amount of unrelinquished connections within our postgres and Metabase pools. We'd like to better manage those numbers, if possible.

## Changes Proposed

- Update default Spring Profile to limit the maximum pool size for Hikari to 5, and the maximum lifetime for connection retention to 60 seconds.
- Update testing Spring Profile to limit the maximum pool size for Hikari to 5, and the maximum lifetime for connection retention to 60 seconds.
- Add variable for MB_APPLICATION_DB_MAX_CONNECTION_POOL_SIZE to limit Metabase connections to 4.
- Add variable for MB_JDBC_DATA_WAREHOUSE_MAX_CONNECTION_POOL_SIZE to limit Metabase JDBC connections to 4.

## Additional Information

- Load testing via unit test runs (100 workflows per minute, approximately) shows that 5 connections seems to be the sweet spot for sustainability. The application starts to choke at 3. We're running a total of 10 connections right now.
- We don't currently set a maximum lifetime value for idle connections. Azure Postgres has this set to an infinite value, and Hikari defaults to 30 minutes. We will want to monitor after these changes go in, just to ensure we are not prematurely closing a connection.

## Testing Methodology

- Hikari changes were instrumented and tested by using repeated local load test runs on isolated hardware with dedicated memory and CPU reservations. Timings were checked for consistency, and the absence of errors was verified at the currently set levels.
- Metabase changes were verified through the use of a local Docker container and manual log inspection.

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/main/resources/application.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ spring:
     username: simple_report_app
     password: api123
     url: jdbc:postgresql://localhost:${SR_DB_PORT:5432}/simple_report
+    hikari:
+      maximum-pool-size: 5 # Maximum number of connections to which Hikari should be limited.
+      max-lifetime: 60000 # Maximum lifetime for a connection to be retained in the pool, in milliseconds, once it is closed.
```
Comment (emmastephenson) [inline backend/src/main/resources/application.yaml:12]: If the max right now is 30 minutes, I'm wondering if we should bump this to a more conservative threshold first before dropping all the way to 60s. Maybe let's try 10 or 15 minutes first, then drop lower if we don't see issues with that. @jdorothy your thoughts?

Comment (jdorothy) [reply backend/src/main/resources/application.yaml:12]: So my understanding is that these settings are effectively the same, in that a PG connection can be recreated in milliseconds, but I agree that maybe `max-lifetime: 600000` is the way to go here _just in case_ the connection re-creation overhead matters. I think optimizing these settings is going to be a work in progress.

Comment (rin-skylight) [reply backend/src/main/resources/application.yaml]: I'm more than happy to make that change!

**Review by nickclyde — APPROVED** [review]
Looks great, thanks for working this out. Hope this helps to prevent the random DB CPU spikes we've been seeing!

**Pull request merged**